Embedded Software/TelemetryControlUnit/src/main.cpp

```diff
@@ -40,89 +40,48 @@
 #define BAUD_RATE 1000000U // 250 kbps 
 #define MAX_MB_NUM 16 // maximum number of CAN mailboxes to use 
 
-#define MAX_BUFFERED_MESSAGES 50 // max number of buffered CAN messages before logging to SD card
 #define MIN_LOG_FREQUENCY 1000 // the max time length between logs (in ms)
 
 #define ACCEL_HUMID_LOG_FREQUENCY 100 // time between logging accel/humid data
 #define ACCEL_LOG_ID 0x300
 #define HUMID_LOG_ID 0x301
 
-typedef struct {
-  char timestamp[25]; // timestamp in YYYY-MM-DDT00:00:00.000Z format
-  uint32_t id;
-  uint8_t length;
-  uint8_t dataBuf[8]; // max number of bytes is 8
-} message_format_t;
 
 FlexCAN_T4<CAN1, RX_SIZE_256, TX_SIZE_16> myCan; // main CAN object
+WDT_T4<WDT1> wdt;
 
-File logFile; // file logging object
+bool loggable = false;
 
-// CAN Ids of the messages to log to SD card (only considered if LOG_ALL is 0)
-const uint32_t LOG_IDS[] = {0x001, 0x002, 0x003, 0x004, 0x0A0, 0x0A1, 0x0A2, 0x0A5, 0x0A6, 0x0A7, 0x0AA, 0x0AB, 0x0AC, 0x202};
-const int NUM_IDS = 14;
-
-// Logging information (use 2 buffers to prevent overwrites during logging delays)
-int buf1Length = 0;
-int buf2Length = 0;
-message_format_t messageBuf1[MAX_BUFFERED_MESSAGES]; 
-message_format_t messageBuf2[MAX_BUFFERED_MESSAGES]; 
-bool usingBuf1 = true; // true when using buf1, false when using buf2
-uint32_t lastLogTime = 0;
-int fileNum = 0; // current file number
-char fileName[16]; // format is log-0.txt (can support files up to number 99999999 as there can be 16 chars)
 
-// Timing information
-uint32_t startUpTimeMillis;
-uint32_t startUpTimeRTC;
-bool useRTC = false; // Default is to use system millis() time
+// CAN Ids of the messages to log to SD card (only considered if LOG_ALL is 0)
+const uint32_t LOG_IDS[] = {0x001, 0x002, 0x003, 0x004, 0x0A0, 0x0A1, 0x0A2, 0x0A5, 0x0A6, 0x0A7, 0x0AA, 0x0AB, 0x0AC, 0x202, 0xC0};
+const int NUM_IDS = sizeof(LOG_IDS) / sizeof(uint32_t);
 
 
 // function declarations
 int sendMessage(uint32_t id, uint8_t len, const uint8_t *buf); 
 void incomingCANCallback(const CAN_message_t &msg);
-bool SDWrite();
-void bufferMessage(uint32_t id, uint8_t len, const uint8_t *buf);
-void getRealTimestamp(char *timestamp);
-void getRelativeTimestamp(char *timestamp);
 void logSensorData();
 
-/**
- * @brief Wrapper function to pass to time sync function
- * 
- * @return time_t 
- */
-time_t getTeensy3Time()
-{
-  return Teensy3Clock.get();
-}
 
 /**
  * @brief Init serial console, CAN bus, and brake switch digital pins
  * 
  */
 void setup() {
+#ifdef DEBUG
   Serial.begin(9600); 
```

Comment:
Does it significantly speed up the code if you don't do Serial.begin(), like why does that have to be a debug conditional?

Reply:
To be honest probably not. I added that when we were debugging the weird logging errors so thought it could be an issue. I'll probably take it out soon.